java/ql/src/experimental/Security/CWE/CWE-691/InsecureClassLoading.qhelp

```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+
+  <overview>
+    <p>
+      Android Contexts that depend on external inputs are potentially vulnerable to code execution attacks.          
+      This query detects the scenarios where an attacker controlled input is used to create a new Android <code> Context</code>. 
+    </p>
+  </overview>
+
+  <recommendation>
+    This can be avoided in two ways. 
+    <ol>
+      <li> 
+        To avoid arbitrary code execution, the application should sanitize all external inputs.
+      </li>
+      <li> Changing the application architecture to prevent unintended access to sensitive components.</li>
```

Comment:
Again be more specific -- what does a good architecture look like here?

Reply:
This is difficult to explain in a `qhelp`. I have included a good example now though. PTAL.